pymeasure/instruments/attocube/anc300.py

```diff
@@ -226,18 +237,51 @@ class ANC300Controller(Instrument):
         "getcser", """ Serial number of the controller board """
     )
 
-    def __init__(self, host, axisnames, passwd, query_delay=0.05, **kwargs):
-        kwargs['query_delay'] = query_delay
+    _reg_value = re.compile(r"\w+\s+=\s+([\w\.]+)")
+
+    def __init__(
+        self,
+        adapter=None,
+        name="attocube ANC300 Piezo Controller",
+        axisnames="",
+        passwd="",
+        query_delay=0.05,
+        **kwargs,
+    ):
+        if adapter is None:
+            adapter = self.handle_deprecated_host_arg(kwargs)
+
+        self.query_delay = query_delay
+        self.termination_str = "\r\n"
+
         super().__init__(
-            AttocubeConsoleAdapter(host, 7230, passwd, **kwargs),
-            "attocube ANC300 Piezo Controller",
+            adapter,
+            name,
             includeSCPI=False,
+            read_termination=self.termination_str,
+            write_termination=self.termination_str,
             **kwargs
         )
+
         self._axisnames = axisnames
         for i, axis in enumerate(axisnames):
             setattr(self, axis, Axis(self, i + 1))
 
+        self.wait_for()
+        # clear messages sent upon opening the connection,
+        # this contains some non-ascii characters!
+        self.adapter.connection.clear()
+        # should be self.adapter.flush_read_buffer() after PR 836
```

Comment:
Yes, we should do it in the order #837 , #836 and then this here. Therefore, let us keep all three PRs in a mergeable state:
```suggestion
        self.adapter.flush_read_buffer()
```

Reply:
Please retrigger CI after merging the respective "preceding" PRs, so that we know the tests pass. It won't rerun automatically after merging another PR afaik.

Reply:
Ok. How can I retrigger CI?

Reply:
Look at the details of one of the jobs below, there should be a "Re-run all jobs" button on the top right.

Reply:
oh wait. I actually don't know if that computes the sha's to use anew, or just runs the same tests with the same merge commit from before...

Reply:
Ah, in principle I think in the PR management section below, there _should_ appear a "Update branch" button if it detects new changes in the base branch.

Reply:
Thanks